lms/djangoapps/django_comment_client/permissions.py

```diff
@@ -5,34 +5,30 @@
 import logging
 from types import NoneType
 from django.core import cache
+
+from request_cache.middleware import RequestCache
 from lms.lib.comment_client import Thread
 from opaque_keys.edx.keys import CourseKey
 
+from django_comment_common.models import all_permissions_for_user_in_course
+
 CACHE = cache.get_cache('default')
 CACHE_LIFESPAN = 60
 
 
-def cached_has_permission(user, permission, course_id=None):
-    """
-    Call has_permission if it's not cached. A change in a user's role or
-    a role's permissions will only become effective after CACHE_LIFESPAN seconds.
-    """
-    assert isinstance(course_id, (NoneType, CourseKey))
-    key = u"permission_{user_id:d}_{course_id}_{permission}".format(
-        user_id=user.id, course_id=course_id, permission=permission)
-    val = CACHE.get(key, None)
-    if val not in [True, False]:
-        val = has_permission(user, permission, course_id=course_id)
-        CACHE.set(key, val, CACHE_LIFESPAN)
-    return val
-
-
 def has_permission(user, permission, course_id=None):
     assert isinstance(course_id, (NoneType, CourseKey))
-    for role in user.roles.filter(course_id=course_id):
-        if role.has_permission(permission):
-            return True
-    return False
+    request_cache_dict = RequestCache.get_request_cache().data
+    cache_key = "django_comment_client.perimissions.has_permission.all_permissions.{}.{}".format(
+        user, course_id
```

Comment:
I realize I wrote this, but this should probably be `user.id`, instead of relying on the serialization of the user obj?